### PR TITLE
keep updating parconnhealth.txt while in connect loop

### PR DESCRIPTION
**Pull request by bill1600**
keep updating the parconnhealth.txt file if we are reconnecting.
Also, make sure 403 error goes through loop and updates parconnhealth.txt

### CODE REVIEW

File: src/connection.c [hunk 1]
```diff
@@ -451,7 +451,7 @@ int wait_connection_ready (create_connection_ctx_t *ctx)
     memset (cfg->webpa_auth_token, 0, sizeof(cfg->webpa_auth_token));
     ParodusError("Received Unauthorized response with status: %d\n", wait_status);
 	OnboardLog("Received Unauthorized response with status: %d\n", wait_status);
-    return WAIT_ACTION_RETRY;
+    return WAIT_FAIL;
```
Comment (shilpa24balaji) [inline src/connection.c]: Why is this change needed?

File: src/conn_interface.c [hunk 2]
```diff
@@ -222,5 +222,6 @@ void createSocketConnection(void (* initKeypress)())
 
 void shutdownSocketConnection(void) {
    g_shutdown = true;
+   reset_interface_down_event ();
```
Comment (shilpa24balaji) [inline src/conn_interface.c]: is this change needed?

**Comment by bill1600** [comment]
The latest version of connection.c has a test for interface down in it.
This line will get it out of that wait.

On 403 errors, we are treating them the same a redirects.  But this means they don't go through back off retry,
and so won't update the parconnhealth.txt file if you keep getting 403s.
That happened to me yesterday when I had the wrong issuer url.
________________________________
